app/PaymentDrivers/Authorize/ChargePaymentProfile.php

```diff
@@ -44,19 +49,42 @@ public function chargeCustomerProfile($profile_id, $payment_profile_id, $amount)
         $profileToCharge->setPaymentProfile($paymentProfile);
 
         $invoice_numbers = '';
-
-        if($this->authorize->payment_hash->data)
-            $invoice_numbers =  collect($this->authorize->payment_hash->data->invoices)->pluck('invoice_number')->implode(',');
+        $taxAmount = 0;
+        $invoiceTotal = 0;
+        $invoiceTaxes = 0;
+
+        if($this->authorize->payment_hash->data) {
+            $invoice_numbers =  collect($this->authorize->payment_hash->data->invoices)->pluck('invoice_number')->implode(",");
+            $invObj = Invoice::whereIn('id', $this->transformKeys(array_column($this->authorize->payment_hash->invoices(), 'invoice_id')))
+                ->withTrashed()->get()->pluck('amount', 'total_taxes')->all();
```

Comment:
for readability it may be best just to use ->get()

and then when iterating use the properties

->amount
->total_taxes

Reply:
Ack